corehq/apps/custom_data_fields/tests/test_form.py

```diff
@@ -0,0 +1,64 @@
+from django.test import SimpleTestCase
+from unittest.mock import patch
+
+from corehq.apps.custom_data_fields.edit_entity import CustomDataEditor
+from corehq.apps.custom_data_fields.models import Field
+
+from corehq.apps.users.views.mobile.custom_data_fields import (
+    UserFieldsView,
+    WebUserFieldsView,
+    CommcareUserFieldsView,
+)
+from corehq.apps.locations.views import LocationFieldsView
+
+
+class FieldsViewMixin:
+
+    def create_field(self, slug='test_field', label='Test Field', is_required=False, required_for=None,
+                     choices=None, regex=None, regex_msg=None, upstream_id=None):
+        if not required_for:
+            required_for = []
+        if not choices:
+            choices = ["option1", "option2"]
+        return Field(
+            slug=slug,
+            label=label,
+            is_required=is_required,
+            required_for=required_for,
+            choices=choices,
+            regex=regex,
+            regex_msg=regex_msg,
+            upstream_id=upstream_id,
+        )
+
+    _create_field = create_field
+
+
+class TestCustomDataEditorFormFields(SimpleTestCase, FieldsViewMixin):
+    def setUp(self):
+        self.domain = 'test-domain'
+
+    @patch('corehq.apps.custom_data_fields.edit_entity.CustomDataEditor.init_form')
+    def test_field_is_required(self, mock_init_form):
```

Comment:
Same with this one - I think it should be 3 different tests (maybe put the _editor and _field vars in the setUp)

Reply:
good call [8dd8f94](https://github.com/dimagi/commcare-hq/pull/35218/commits/8dd8f94b8bc4a4b453438d23bbee8391b8838d61)